### PR TITLE
Update build instructions in README.md

### PR DESCRIPTION
**Pull request by DontBelieveMe**
The old instructions where a bit confusing as they seemed to be mixing up platform specific steps.
These new instructions are not necessarily perfect or the most detailed, but they are a bit clearer

@apachano requesting your review as this is more documentation related and you seem to be covering that kind of thing.

### CODE REVIEW

**Comment by apachano** [comment]
Here is the result of following the Linux instructions, testing Mac next.
image

**Comment by DontBelieveMe** [comment]
Ah yeah, just reproduced this with WSL with CMake 3.10.2.
This is because `-S` is only introduced as the officially supported "Source Directory" flag in 3.13.0.
I changed this because `-H.` is not a documented or officially supported way of doing it either (and in fact I think it has changed meaning recently).

Maybe just having
```
mkdir Build
cmake ..
cd ..
```
as that appears to be a more stable way to do it?

**Comment by DontBelieveMe** [comment]
Another thing we should probably do is use our documented build instructions in the CI config as how we build - currently it does the old thing and uses `cmake -H. -BBuild` instead of the way proposed above.

**Comment by SuperFola** [comment]
Or just cmake . -BBuild

**Comment by vyomfadia** [comment]
I think we should just require 3.13.0 or higher and use the *actually official* -S argument, I've tested on Windows and Linux so far, and it seems to work just fine. If @apachano could possibly get anything higher on Mac and test it, I think it would work just fine.

**Review by apachano — CHANGES_REQUESTED** [review]
For both Mac and Linux, the executable is located at Build/QuartzSandbox/QuartzSandbox

File: README.md [hunk 1]
```diff
@@ -14,22 +14,27 @@
...
+  - At this point you should be able to run, since the project should have already been
+    built in step 2. above. You can always build the traditional way with Visual Studio.
+  - And voila, all done. Now you should be able to run the project!
 
-  1. `mkdir build && cd build`
-  2. `cmake ..`
-  3. `make`
+### Linux, Mac OS X, MSYS
+ 
+  - Navigate to the `Build/` folder and run `./QuartzSandbox` to run the executable.
```
Comment (apachano) [inline README.md]: So they need to navigate to Build/<project>/ then run ./<project>
In this case CD Build/QuatzSandbox/ then ./QuartzSandbox

Comment (DontBelieveMe) [reply README.md]: Ah damn I did wonder if I had got this the right way round :D Trying to write linux instructions from windows was always going to end badly.

I'll fix this now

**Review by apachano — APPROVED** [review]
Lol, I was actually good with `cmake -S. -BBuild` once I got it working. Linux being the only barrier, the default Ubuntu PPA is on CMake 3.10 (for 19), the difference was just adding the official KitWare PPA to get 3.15

All the same I am okay with this, everything makes sense and is easy to follow

**Pull request merged**